netket/utils/numbers.py

```diff
@@ -0,0 +1,38 @@
+from numbers import Number
+
+from typing import Union, Any
+from plum import dispatch
+
+import numpy as np
+import jax
+import jaxlib
+
+ArrayT = {np.ndarray, jaxlib.xla_extension.DeviceArray, jax.core.Tracer}
+
+
+@dispatch.annotations()
+def dtype(x: Number):
+    return type(x)
+
+
+@dispatch.annotations()
+def dtype(x: ArrayT):
+    return x.dtype
+
+
+dispatch.annotations()
```

Comment:
What is this lonely line doing here?

Reply:
Ops, little red riding line got lost in the forest